src/Controller/ArticleController.php

```diff
@@ -85,7 +85,7 @@ public function add(): string
             $this->checkUploadFile($_FILES, $errors);
 
             if (empty($errors)) {
-                $newArticle = $_POST;
+                $newArticle = array_map('trim', $_POST);
```

Comment:
use trim function BEFORE to check your form (empty spaces aren't considered as empty field)
Then put this newArticle on your validation method and return it to twig, (to avoid errors or duplicate checks you can initialize newArticle as an empty array at the beginning of your method)